src/h2/utilities.py

```diff
@@ -105,27 +105,26 @@ def extract_method_header(headers):
 
 def is_informational_response(headers):
     """
-    Searches a header block for a :status header to confirm that a given
+    Searches headers list for a :status header to confirm that a given
     collection of headers are an informational response. Assumes the header
-    block is well formed: that is, that the HTTP/2 special headers are first
-    in the block, and so that it can stop looking when it finds the first
-    header field whose name does not begin with a colon.
+    are well formed and encoded as bytes: that is, that the HTTP/2 special
+    headers are first in the block, and so that it can stop looking when it
+    finds the first header field whose name does not begin with a colon.
 
-    :param headers: The HTTP/2 header block.
+    :param headers: The HTTP/2 headers.
     :returns: A boolean indicating if this is an informational response.
     """
     for n, v in headers:
-        # If we find a non-special header, we're done here: stop looping.
+        if not isinstance(n, bytes) or not isinstance(v, bytes):
+            raise ProtocolError(f"header not bytes: {n=:r}, {v=:r}")  # pragma: no cover
 
-        if n and n[0] != SIGIL:
+        if not n.startswith(b':'):
```

Comment:
Why this change? (And the other `.startswith()` below from my `n and n[0] == SIGIL` version?) I'm quite sure they are significantly slower as they involve a function call and are not optimized for 1-byte look up.